python/core/qgsstatisticalsummary.sip

```diff
@@ -1,214 +1,299 @@
+ or via setStatistics.
+ :rtype: int
+%End
 
-    /** Returns minority of values. The minority is the value with least occurrences in the list
-     * This is only calculated if Statistic::Minority has been specified in the constructor
-     * or via setStatistics.
-     * @see majority
-     */
     double minority() const;
+%Docstring
+ Returns minority of values. The minority is the value with least occurrences in the list
+ This is only calculated if Statistic.Minority has been specified in the constructor
+ or via setStatistics. A NaN value may be returned if the minority cannot
+ be calculated.
+.. seealso:: majority
+ :rtype: float
+%End
 
-    /** Returns majority of values. The majority is the value with most occurrences in the list
-     * This is only calculated if Statistic::Majority has been specified in the constructor
-     * or via setStatistics.
-     * @see minority
-     */
     double majority() const;
+%Docstring
+ Returns majority of values. The majority is the value with most occurrences in the list
+ This is only calculated if Statistic.Majority has been specified in the constructor
+ or via setStatistics. A NaN value may be returned if the majority cannot
+ be calculated.
+.. seealso:: minority
+ :rtype: float
+%End
 
-    /** Returns the first quartile of the values. The quartile is calculated using the
-     * "Tukey's hinges" method.
-     * @see thirdQuartile
-     * @see interQuartileRange
-     */
     double firstQuartile() const;
+%Docstring
+ Returns the first quartile of the values. The quartile is calculated using the
+ "Tukey's hinges" method. A NaN value may be returned if the first quartile cannot
+ be calculated.
+.. seealso:: thirdQuartile
+.. seealso:: interQuartileRange
+ :rtype: float
+%End
 
-    /** Returns the third quartile of the values. The quartile is calculated using the
-     * "Tukey's hinges" method.
-     * @see firstQuartile
-     * @see interQuartileRange
-     */
     double thirdQuartile() const;
+%Docstring
+ Returns the third quartile of the values. The quartile is calculated using the
+ "Tukey's hinges" method. A NaN value may be returned if the third quartile cannot
+ be calculated.
+.. seealso:: firstQuartile
+.. seealso:: interQuartileRange
+ :rtype: float
+%End
 
-    /** Returns the inter quartile range of the values. The quartiles are calculated using the
-     * "Tukey's hinges" method.
-     * @see firstQuartile
-     * @see thirdQuartile
-     */
     double interQuartileRange() const;
+%Docstring
+ Returns the inter quartile range of the values. The quartiles are calculated using the
+ "Tukey's hinges" method. A NaN value may be returned if the IQR cannot
+ be calculated.
+.. seealso:: firstQuartile
+.. seealso:: thirdQuartile
+ :rtype: float
+%End
 
-    /** Returns the friendly display name for a statistic
-     * @param statistic statistic to return name for
-     */
     static QString displayName( Statistic statistic );
```

Comment:
you might need to add `QgsStatisticalSummary`in front of `Statistic` (in the header)